plugins/pastefromword/filter/default.js

```diff
@@ -354,41 +343,128 @@
 
 				},
 				'td': function( element ) {
-
 					var ascendant = element.getAscendant( 'table' ),
-						tdBorders =  ascendant._tdBorders,
-						borderStyles = [ 'border', 'border-top', 'border-right', 'border-bottom', 'border-left' ],
-						ascendantStyle = tools.parseCssText( ascendant.attributes.style );
+						allowedBorderTypes = [ 'border-top', 'border-right', 'border-bottom', 'border-left' ],
+						borderModifiers = [ 'style', 'width', 'color' ],
+						ascendantStyle = tools.parseCssText( ascendant.attributes.style, true );
 
 					// Sometimes the background is set for the whole table - move it to individual cells.
-					var background = ascendantStyle.background || ascendantStyle.BACKGROUND;
+					var background = ascendantStyle.background;
 					if ( background ) {
 						Style.setStyle( element, 'background', background, true );
 					}
 
-					var backgroundColor = ascendantStyle[ 'background-color' ] || ascendantStyle[ 'BACKGROUND-COLOR' ];
+					var backgroundColor = ascendantStyle[ 'background-color' ];
 					if ( backgroundColor ) {
 						Style.setStyle( element, 'background-color', backgroundColor, true );
 					}
 
-					var styles = tools.parseCssText( element.attributes.style );
+					var styles = tools.parseCssText( element.attributes.style, true ),
+						borderStyles = styles.border ? CKEDITOR.tools.style.parse.border( styles.border ) : {};
 
-					for ( var style in styles ) {
-						var temp = styles[ style ];
-						delete styles[ style ];
-						styles[ style.toLowerCase() ] = temp;
-					}
+					for ( var i = 0; i < allowedBorderTypes.length; i++ ) {
```

Comment:
You use `CKEDITOR.tools.array.forEach` just right after this loop. Won't be better to use also here?